server/src/interfaces/session.interface.ts

```diff
@@ -2,10 +2,12 @@ import { SessionEntity } from 'src/entities/session.entity';
 
 export const ISessionRepository = 'ISessionRepository';
 
+type E = SessionEntity;
```

Comment:
That was necessary lmao

Reply:
I was considering making all interfaces implement a `BaseInterface` that has the correct typings for `create`, `update`, `delete`, and `get`. It would be something like `BaseInterface<E>` and then the methods would indeed reference `E`.